pkgs/applications/misc/inochi2d/default.nix

```diff
@@ -0,0 +1,111 @@
+in
+{
+  inochi-creator = mkGeneric rec {
+    pname = "inochi-creator";
+    appname = "Inochi Creator";
+    version = "0.8.3";
+
+    src = fetchFromGitHub {
+      owner = "Inochi2D";
+      repo = "inochi-creator";
+      rev = "v${version}";
+      hash = "sha256-yyjsiyxXdETykTzRol9IIsOxIpjRsvoUi2+6SidvxQ8=";
+    };
+
+    # Note: there exists a custom fork of dcv (dcv-i2d) which fixes some issues with the original dcv, and this fork is fetched by the upstream github workflow.
+    # However, it seems like it's not actually used anywhere in the repo, so I didn't include it here.
+    dubDeps = ./creator-deps.nix;
+
+    # Upstream says that we must change the bug tracker URL to not point to the upsteam bug tracker
+    patches = [ ./support-url.patch ];
+
+    postPatch = ''
+      # set custom translation directory
+      substituteInPlace source/creator/core/path.d \
+          --replace-fail 'incGetAppLocalePathExtra() {' "incGetAppLocalePathExtra() { return \"$out/share/inochi-creator/i18n\";"
+
+      # don't use path for differentiating between translation locales (the length of the store path looks bad)
+      substituteInPlace source/creator/core/i18n.d \
+          --replace-fail 'prevEntry.path' 'prevEntry.code'
+
+      substituteInPlace $DUB_DEPS/i2d-imgui/dub.sdl \
+          --replace-fail 'libs "sdl2"' 'libs "SDL2"'
+
+      cp -r --no-preserve=all ${cimgui-src}/* $DUB_DEPS/i2d-imgui/deps/cimgui
+      sed -i $DUB_DEPS/i2d-imgui/deps/CMakeLists.txt \
+          -e '/PullSubmodules/d' \
+          -e 's/''${cimgui_SUBMOD_DIR}/cimgui/g'
+    '';
+
+    # generate translations
+    preBuild = ''
+      chmod +x ./gentl.sh
+      ./gentl.sh
+    '';
+
+    # copy translations
+    preInstall = ''
+      install -Dm644 out/*.mo -t $out/share/inochi-creator/i18n
+    '';
+
+    meta = {
+      broken = stdenv.isDarwin; # darwin has slightly different build steps
+      changelog = "https://github.com/Inochi2D/inochi-creator/releases/tag/${src.rev}";
+      description = "An open source editor for the Inochi2D puppet format";
+    };
+  };
+
+  inochi-session = mkGeneric rec {
+    pname = "inochi-session";
+    appname = "Inochi Session";
+    version = "0.8.3";
+
+    src = fetchFromGitHub {
+      owner = "Inochi2D";
+      repo = "inochi-session";
+      rev = "v${version}";
+      hash = "sha256-yq/uMWEeydZun07/7hgUaAw3IruRqrDuGgbe5NzNYxw=";
+    };
+
+    dubDeps = ./session-deps.nix;
+
+    postPatch = ''
+      cp -r --no-preserve=all ${cimgui-src}/* $DUB_DEPS/i2d-imgui/deps/cimgui
+      sed -i $DUB_DEPS/i2d-imgui/deps/CMakeLists.txt \
+          -e '/PullSubmodules/d' \
+          -e 's/''${cimgui_SUBMOD_DIR}/cimgui/g'
+    '';
+
+    postInstall = ''
+      install -Dm755 out/cimgui.so -t $out/lib
+    '';
+
+    preFixup = ''
+      patchelf $out/bin/inochi-session --add-needed $out/lib/cimgui.so
+    '';
+
+    dontStrip = true; # symbol lookup error: undefined symbol: , version
```

Comment:
Try

```suggestion
    preFixup = ''
      patchelf $out/bin/inochi-session --add-needed cimgui.so --add-rpath $out/lib
    '';
```

Reply:
This doesn't work, as I have said previously. Removing `dontStrip` and adding `--add-rpath` caused it to segfault immediately, without an error message. `LD_DEBUG=all` barely showed anything being loaded and `ldd` doesn't even recognize it as an executable.

Reply:
Hm that's annoying. I'd really rather not disable stripping. I'll experiment around perhaps.

Reply:
Here's the difference between stripping and not stripping according to `LD_DEBUG=all`

[no-strip.txt](https://github.com/NixOS/nixpkgs/files/14322519/no-strip.txt)
[strip.txt](https://github.com/NixOS/nixpkgs/files/14322521/strip.txt)

AFAICT, the stripped one gets stuck in an infinite recursion trying to find an empty symbol. Weird